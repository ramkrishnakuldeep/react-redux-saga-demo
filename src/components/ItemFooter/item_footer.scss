footer {
    position: fixed;
    height: 60px;
    width: 100%;
    bottom: 0;
    background: maroon;
    display: flex;
    align-items: center;
    justify-content: space-between;
    color: white;

    .price {
        font-size: 2rem;
    }

    .add_to_cart {
        border: 0;
        height: 45px;
        min-width: 100px;
        color: white;
        background: forestgreen;
    }

    div:first-child {
        margin-left: 20px;
    }

    div:last-child {
        margin-right: 20px;
    }
}