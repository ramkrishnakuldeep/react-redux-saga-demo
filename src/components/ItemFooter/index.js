import React from 'react';
import { itemDetails } from '../../propTypes';
import './item_footer.scss';

const ItemFooter = ({ item }) => {
    return (
        <footer>
            <div>
                <span className="price"> &#165; {item.price} </span>
                <span> { item.shipping_fee} </span>
            </div>
            <div>
                <button className="add_to_cart">
                    Add to cart
                </button>
            </div>
        </footer>
    );
};

ItemFooter.defaultProps = {
    item: {}
};

ItemFooter.propTypes = { item: itemDetails };

export default ItemFooter;
