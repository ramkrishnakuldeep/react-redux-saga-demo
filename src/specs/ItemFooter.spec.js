import React from 'react';
import { shallow } from '../enzyme';
import ItemFooter from '../components/ItemFooter';

describe('ItemFooter component', () => {
    it('initial render', () => {
        const component = shallow(<ItemFooter />);
        expect(component).toMatchSnapshot();
    });
});
