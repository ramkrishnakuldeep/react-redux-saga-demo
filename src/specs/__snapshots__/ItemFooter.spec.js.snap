// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemFooter component initial render 1`] = `
<footer>
  <div>
    <span
      className="price"
    >
       Â¥ 
       
    </span>
    <span>
       
       
    </span>
  </div>
  <div>
    <button
      className="add_to_cart"
    >
      Add to cart
    </button>
  </div>
</footer>
`;
